### PR TITLE
ZER-3 Calculator: create a main page (FE)

### PR DESCRIPTION
**Pull request by Har4enkoO**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: app/controllers/pages_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class PagesController < ApplicationController
```
Comment (maksymst) [inline app/controllers/pages_controller.rb:1]: Rename to CalculatorsController

File: app/helpers/pages_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module PagesHelper
```
Comment (maksymst) [inline app/helpers/pages_helper.rb:1]: See above

File: app/controllers/pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+  def home
```
Comment (maksymst) [inline app/controllers/pages_controller.rb:2]: `show` action

File: db/schema.rb [hunk 4]
```diff
@@ -10,11 +10,19 @@
...
   enable_extension "plpgsql"
 
+  create_table "calculators", force: :cascade do |t|
+    t.uuid "uuid", null: false
+    t.string "name"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.index ["uuid"], name: "index_calculators_on_uuid", unique: true
+  end
+
```
Comment (maksymst) [inline db/schema.rb:25]: update your branch. This is already in master

File: test/controllers/pages_controller_test.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class PagesControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (maksymst) [inline test/controllers/pages_controller_test.rb:7]: Generate Rspec test instead of this one

**Review by maksymst — CHANGES_REQUESTED** [review]
Also please attach a screenshot of the page

File: app/assets/stylesheets/cutom.css.scss [hunk 6]
```diff
@@ -0,0 +1,67 @@
+@import 'bootstrap/dist/css/bootstrap';
+
```
[the remark this answers is not in the document]
Comment (Har4enkoO) [reply app/assets/stylesheets/cutom.css.scss:2]: Thanks, done.

File: app/controllers/pages_controller.rb (hunk 1 above)
Comment (Har4enkoO) [reply app/controllers/pages_controller.rb:1]: Thanks, done.

File: app/helpers/pages_helper.rb (hunk 2 above)
Comment (Har4enkoO) [reply app/helpers/pages_helper.rb:1]: Thanks, done.

File: db/schema.rb (hunk 4 above)
Comment (Har4enkoO) [reply db/schema.rb:25]: Thanks, done.

**Comment by Har4enkoO** [comment]
Screenshot from 2021-04-21 10-05-59

File: app/assets/stylesheets/pages.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the pages controller here.
```
Comment (maksymst) [inline app/assets/stylesheets/pages.scss:1]: rename this one too

File: app/controllers/calculators_controller.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class CalculatorsController < ApplicationController
+  def calculator
```
Comment (maksymst) [inline app/controllers/calculators_controller.rb:2]: it should be `show` action

File: app/views/calculators/calculator.html.erb [hunk 9]
```diff
@@ -0,0 +1,41 @@
+<h1 class="display-4 text-center mt-5 mb-5">
```
Comment (maksymst) [inline app/views/calculators/calculator.html.erb:1]: rename this template according to controller action name

File: app/views/layouts/_navigation.html.erb [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    <a href="#" class="tab">TAB1</a>
+    <a href="#" class="tab">TAB2</a>
+    <a href="#" class="tab">TAB3</a>
+    <a href="#" class="tab">TAB4</a>
+  </div>
+  <div class="donate">
+    <button type="button" class="btn btn-success">
+      <a href="https://zerowastelviv.org.ua/pidtrymaty/" class="donate-btn"
+        >DONATE</a
+      >
```
Comment (maksymst) [inline app/views/layouts/_navigation.html.erb:15]: is there any reason of formatting like this?

File: config/routes.rb [hunk 11]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  root 'calculators#calculator'
```
Comment (maksymst) [inline config/routes.rb:4]: let's not set the `root`. later we may have some landing page, but for now, we have only a specific calculator page with URI like `calculators/diapers-calculator`

File: spec/views/calculators/calculator.html.erb_spec.rb [hunk 12]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Calculator, type: :view do
+  describe "rendering text directly" do
+    it "displays the given text" do  
+      render :calculator => "This is directly rendered"  
+      rendered.should contain("directly rendered")
+    end
+  end
+end
```
Comment (maksymst) [inline spec/views/calculators/calculator.html.erb_spec.rb:10]: you can skip this test. we will cover this funcitonality later

File: test/controllers/calculators_controller_test.rb [hunk 13]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
+
+class CalculatorsControllerTest < ActionDispatch::IntegrationTest
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (maksymst) [inline test/controllers/calculators_controller_test.rb:7]: remove it for now

**Review by maksymst — CHANGES_REQUESTED** [review]

File: spec/views/calculators/calculator.html.erb_spec.rb (hunk 12 above)
Comment (Har4enkoO) [reply spec/views/calculators/calculator.html.erb_spec.rb:10]: Thanks, done.

File: test/controllers/calculators_controller_test.rb (hunk 13 above)
Comment (Har4enkoO) [reply test/controllers/calculators_controller_test.rb:7]: Thanks, done.

File: app/assets/stylesheets/pages.scss (hunk 7 above)
Comment (Har4enkoO) [reply app/assets/stylesheets/pages.scss:1]: Thanks, done.

File: app/views/layouts/_navigation.html.erb (hunk 10 above)
Comment (Har4enkoO) [reply app/views/layouts/_navigation.html.erb:15]: Thanks, done.

File: app/views/calculators/calculator.html.erb (hunk 9 above)
Comment (Har4enkoO) [reply app/views/calculators/calculator.html.erb:1]: Thanks, done.

File: app/controllers/calculators_controller.rb (hunk 8 above)
Comment (Har4enkoO) [reply app/controllers/calculators_controller.rb]: Thanks, done.

File: config/routes.rb (hunk 11 above)
Comment (Har4enkoO) [reply config/routes.rb:5]: Thanks, done.

File: app/controllers/calculators_controller.rb [hunk 14]
```diff
@@ -0,0 +1,7 @@
+class CalculatorsController < ApplicationController
+
+  def show
+    render "#{params[:id]}"
```
Comment (maksymst) [inline app/controllers/calculators_controller.rb:4]: just render `show` template

**Review by maksymst — CHANGES_REQUESTED** [review]

**Review by maksymst — APPROVED** [review]

**Pull request merged**